packages/ragbits-core/docs/how-to/hybrid-search.md

```diff
@@ -0,0 +1,169 @@
+    "config": {
+        "api_key": "your-api-key",
+        "model": "text-embedding-ada-002"
+    }
+})
+
+# Sparse embedder
+sparse_embedder = SparseEmbedder()
+```
+
+### Step 2: Create Vector Stores for Each Embedder Type
+
+Create separate vector stores for dense and sparse embeddings:
+
+```python
+from ragbits.core.vector_stores.in_memory import InMemoryVectorStore
+from ragbits.core.vector_stores.base import EmbeddingType, VectorStoreOptions
+
+# Vector store for dense embeddings
+dense_store = InMemoryVectorStore(
+    embedder=dense_embedder,
+    embedding_type=EmbeddingType.TEXT,
+    default_options=VectorStoreOptions(k=10)
+)
+
+# Vector store for sparse embeddings
+sparse_store = InMemoryVectorStore(
+    embedder=sparse_embedder,
+    embedding_type=EmbeddingType.TEXT,
+    default_options=VectorStoreOptions(k=10)
+)
+```
+
+### Step 3: Create a Hybrid Search Vector Store
+
+Combine the vector stores using `HybridSearchVectorStore`:
+
+```python
+from ragbits.core.vector_stores.hybrid import HybridSearchVectorStore
+
+hybrid_store = HybridSearchVectorStore(
+    stores=[dense_store, sparse_store],
+    weights=[0.7, 0.3]  # Weights for each store (must sum to 1.0)
+)
+```
+
+### Step 4: Store and Retrieve Entries
+
+Store your entries in both vector stores and then query using the hybrid store:
+
+```python
+import uuid
+from ragbits.core.vector_stores.base import VectorStoreEntry
+
+# Create entries
+entries = [
+    VectorStoreEntry(
+        id=uuid.uuid4(),
+        text="This is a sample document about artificial intelligence",
+        metadata={"category": "AI"}
+    ),
+    VectorStoreEntry(
+        id=uuid.uuid4(),
+        text="Machine learning algorithms can process large datasets",
+        metadata={"category": "ML"}
+    )
+]
+
+# Store entries in both vector stores
+await dense_store.store(entries)
+await sparse_store.store(entries)
+
+# Retrieve similar entries using hybrid search
+results = await hybrid_store.retrieve("AI algorithms for big data")
+
+# Access results
+for result in results:
+    print(f"Text: {result.entry.text}")
+    print(f"Score: {result.score}")
+    print(f"Vector type: {type(result.vector).__name__}")
+```
+
+## Fine-tuning Hybrid Search
+
+You can adjust the weights of each vector store to optimize for your specific use case:
+
+- Higher weight for dense store: Better semantic understanding but might miss exact keyword matches
+- Higher weight for sparse store: Better keyword matching but might miss semantically related content
+- Equal weights: Balanced approach
+
+Experiment with different weights to find the optimal configuration for your data and queries.
+
+## Advanced Hybrid Search Techniques
+
+### Reranking
+
+For more sophisticated hybrid search, you can implement a reranking step:
+
+```python
+from ragbits.core.vector_stores.base import VectorStoreResult
+
+async def rerank_results(query: str, results: list[VectorStoreResult]) -> list[VectorStoreResult]:
```

Comment:
We have a separate reranking solution. There is no reason to revinvent in in this document